seqr/migrations/0052_rnaseqspliceoutlier_rank.py

```diff
@@ -0,0 +1,19 @@
+# Generated by Django 3.2.18 on 2023-06-09 18:00
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('seqr', '0051_auto_20230426_1615'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='rnaseqspliceoutlier',
+            name='rank',
+            field=models.IntegerField(default=0),
```

Comment:
We need this to fail if there is data loaded - setting this default will break behavior for displaying these models, we are relying on this field to be accurate. This needs to be non-nullable with no default - you will need to drop all your data locally to run this

Reply:
It still requires a default value after dropping all local data. And I did a little investigation and found there is a [trick ](https://stackoverflow.com/questions/30885466/how-can-i-resolve-the-django-error-add-a-non-nullable-field) to adding a non-nullable with no default field by two steps: 1) adding a nullable field first, 2) changing it to a non-nullable (by selecting manually handling null values). There will be two migration files. Do you think this approach is acceptable?

Reply:
I think you are misunderstanding how the second option works. If you select "manually handle" it will generate a migration file for you with no default and  a non-nullable field, which is exactly what we want. You then have the ability to manually add your own handling for that,  but you should just not add anything and we will be all set

Reply:
I know "manually handle" is what we want. But we can't simply add a new field to achieve that. We have to add a new field with "null=True" first and do "makemigrations", then remove "null=True" and do "makemigrations" again. We can only select "manually handle" when we do "makemigrations" for the second time.

Reply:
I pushed an update that has two migrations. If there is any RnaSeqSpliceOutlier data, the migration will fail because the data is non-nullable. But the migrations will be successful if there are no existing RnaSeqSpliceOutlier data.

Reply:
Why would there be 2 migrations? This needs to be one migration that fails if there is data, and since there is not any data we are fine

Reply:
I know how to reverse the migrations. Doing it in one migration is only possible by providing a default value. The two migrations approach is the trick I mentioned above.

Reply:
It also works by merging the two migration files manually.